src/transports.jl

```diff
@@ -195,3 +195,33 @@ read!(t::TFileTransport, buff::Vector{UInt8}) = read!(t.handle, buff)
 read(t::TFileTransport, UInt8) = read(t.handle, UInt8)
 write(t::TFileTransport, buff::Vector{UInt8}) = write(t.handle, buff)
 write(t::TFileTransport, b::UInt8) = write(t.handle, b)
+
+# Thrift Http Transport
+mutable struct THttpClient <: Thrift.TTransport
+    url::String
+    headers::Dict
+    rbuff::IOBuffer
+    wbuff::IOBuffer
+    THttpClient(url::String, headers::Dict) = new(url, headers, PipeBuffer(), PipeBuffer())
+end
+
+function flush(t::THttpClient)
+    t.rbuff = PipeBuffer()
+    headers = Dict(
+        "Content-Type" => "application/x-thrift",
+    );
+    for (key, val) in t.headers
+        headers[key] = val
+    end
+    response = HTTP.request("POST", t.url, headers=headers, body=body = take!(t.wbuff))
+    write(t.rbuff, response.body)
```

Comment:
Why not create a IOBuffer using `response.body` and assign to `rbuff`? Will avoid a write and buffer expansion.